indent.go

```diff
@@ -18,130 +19,191 @@ const (
 	replSaveURL = "https://repl.it/save"
 )
 
-type replData struct {
-	id       string
-	language string
-	url      string
-	newURL   string
-	code     string
+type replFile struct {
+	ID      int    `json:"id"`
+	Name    string `json:"name"`
+	Content string `json:"content"`
+	Index   int    `json:"index"`
+
 	indented string
-	errors   string
 }
 
-func extractReplItTag(body *[]byte, tag string) (string, error) {
-	regex := regexp.MustCompile(fmt.Sprintf("%q:%q", tag, "(.*?)"))
-	match := regex.FindSubmatch(*body)
-	if match == nil {
-		return "", fmt.Errorf("não foi possível extrair tag %s", tag)
-	}
-	return string(match[1]), nil
+type replProject struct {
+	SessionID  string              `json:"session_id"`
+	RevisionID string              `json:"revision_id"`
+	EditorText string              `json:"editor_text"`
+	IsProject  bool                `json:"is_project"`
+	Language   string              `json:"language"`
+	Files      map[string]replFile `json:"files"`
+
+	url                string
+	newURL             string
+	editorTextIndented string
+	indentedByUs       bool
 }
 
-func downloadReplIt(url string) (*replData, error) {
-	resp, err := http.Get(url)
+func parseReplItJSON(data []byte) (*replProject, error) {
+	var repl replProject
+	err := json.Unmarshal(data, &repl)
 	if err != nil {
 		return nil, err
 	}
-	defer resp.Body.Close()
-	body, err := ioutil.ReadAll(resp.Body)
 
-	repl := replData{url: url}
+	// Unescaping HTML.
+	repl.Language = strings.ToLower(html.UnescapeString(repl.Language))
+	repl.EditorText = html.UnescapeString(repl.EditorText)
+	for key, file := range repl.Files {
+		file.Name = html.UnescapeString(file.Name)
+		file.Content = html.UnescapeString(file.Content)
+		repl.Files[key] = file
+	}
 
-	var start, end int
-	// Code.
-	if start = bytes.Index(body, []byte("<pre>")); start != -1 {
-		if end = bytes.Index(body, []byte("</pre>")); end != -1 {
-			repl.code = html.UnescapeString(string(body[start+len("<pre>") : end]))
-		}
+	return &repl, nil
+}
+
+func parseReplItDownload(body *[]byte) (*replProject, error) {
+	regex := regexp.MustCompile("REPLIT_DATA = ({.*})</script>")
```

Comment:
MustCompile is normally used for initializations. Since the function returns an error, it wouldn't hurt returning error from regexp.Compile (although that's entirely optional).